### PR TITLE
[WIP] Enhanced createSelect functionality

### LINKED ISSUE #3880: p5.dom createSelect() options cannot be set to disabled or selected
#### Nature of issue?

- [ ] Found a bug
- [x] Existing feature enhancement
- [ ] New feature request

#### Most appropriate sub-area of p5.js?

- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] Events
- [ ] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [ ] WebGL
- [X] Other (specify if possible)

p5.dom

#### p5.js version 

0.9.0

#### Which platform were you using when you encountered this?

- [ ] Mobile/Tablet (touch devices)
- [X] Desktop/Laptop
- [ ] Others (specify if possible)

#### Feature enhancement details:

I'm not sure how to set an attribute like `disabled` or `selected` (default option) on the options of a Select element as `createSelect().option()` returns `undefined`. 

https://p5js.org/reference/#/p5/createSelect

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/option

the following workaround works for `disabled` but not for `selected`
  
```
  const selectPlayer1 = createSelect().id('s1');
  selectPlayer1.position(10, 10);
  selectPlayer1.option('a');
  selectPlayer1.option('b');
  selectPlayer1.option('c');
  // selectPlayer1.changed(onSelectPlayer1);
  const options = selectAll('option', 's1');
  options[1].attribute('disabled', 'disabled');  
```

### PR DESCRIPTION
**Pull request by jatin33**
Resolves #3880 

 Changes: 

Implemented .disable(optionName) method which disables individual option of dropdown element.

#### PR Checklist

- [x] `npm run lint` passes
- [x] [Inline documentation] is included / updated
- [x] [Unit tests] are included / updated
- [ ] [Benchmarks] are included / updated

[Inline documentation]: https://github.com/processing/p5.js/blob/master/contributor_docs/inline_documentation.md
[Unit tests]: https://github.com/processing/p5.js/tree/master/contributor_docs#unit-tests
[Benchmarks]: https://github.com/processing/p5.js/blob/master/contributor_docs/benchmarking_p5.md

### CODE REVIEW

File: src/dom/dom.js [hunk 1]
```diff
@@ -713,6 +713,18 @@ p5.prototype.createSelect = function() {
...
+
+  self.disable = function(value) {
+    if (value !== undefined && typeof value === 'string') {
+      for (let i = 0; i < this.elt.length; i++) {
+        if (this.elt[i].value === value) {
+          this.elt[i].disabled = true;
+        }
+      }
+      return this;
+    }
```
Comment (lmccart) [inline src/dom/dom.js]: should there be an else here that disables the entire element if no option string is specified?

File: src/dom/dom.js [hunk 2]
```diff
@@ -713,6 +713,18 @@ p5.prototype.createSelect = function() {
       }
     }
   };
+
+  self.disable = function(value) {
```
Comment (lmccart) [inline src/dom/dom.js]: can you add documentation for this method?

File: src/dom/dom.js (hunk 1 above)
Comment (jatin33) [reply src/dom/dom.js]: Yes that would be great. I have implemented it in commit 4da60ae

File: src/dom/dom.js (hunk 2 above)
Comment (jatin33) [reply src/dom/dom.js]: Working on it.

Comment (jatin33) [reply src/dom/dom.js]: Added documentation in commit 4136c5c

**Comment by jatin33** [comment]
Below is the screenshot of changes appearing in example docs:
disableOptions

**Comment by jatin33** [comment]
> this looks great. while you're here editing this file, would you mind adding a sentence documenting .option() as well?

Sure, I will do that as well.

**Comment by jatin33** [comment]
Also, do you want me to resolve the conflicts?
I tried resolving conflicts but was unable to finalize them, I guess that is because I don't have write access. Can you take a look at it, please?

**Comment by lmccart** [comment]
I have resolved the conflicts. I don't know why these were there. Please be sure that you are pulling the latest from the p5.js master branch before working. And that your version of node is up to date.

**Comment by jatin33** [comment]
> I have resolved the conflicts. I don't know why these were there. Please be sure that you are pulling the latest from the p5.js master branch before working. And that your version of node is up to date.

Yes, that was my mistake. I will take care of this workflow from now on.

**Pull request merged**